File: docs/breez-sdk/snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/Contacts.kt
package com.example.kotlinmpplib

import breez_sdk_spark.*

class Contacts {
    suspend fun addContact(sdk: BreezSdk) {
        // ANCHOR: add-contact
        val contact = sdk.addContact(AddContactRequest(
            name = "Alice",
            paymentIdentifier = "alice@example.com"
        ))
        // Log.v("Breez", "Contact added: $contact")
        // ANCHOR_END: add-contact
    }

    suspend fun updateContact(sdk: BreezSdk) {
        // ANCHOR: update-contact
        val contactId = "contact-id"
        val contact = sdk.updateContact(UpdateContactRequest(
            id = contactId,
            name = "Alice Smith",
            paymentIdentifier = "alice.smith@example.com"
        ))
        // Log.v("Breez", "Contact updated: $contact")
        // ANCHOR_END: update-contact
    }

    suspend fun deleteContact(sdk: BreezSdk) {
        // ANCHOR: delete-contact
        val contactId = "contact-id"
        sdk.deleteContact(contactId)
        // Log.v("Breez", "Contact deleted")
        // ANCHOR_END: delete-contact
    }

    suspend fun listContacts(sdk: BreezSdk) {
        // ANCHOR: list-contacts
        // List contacts with pagination (e.g., 10 contacts starting from offset 0)
        // Optionally filter by exact name match
        val contacts = sdk.listContacts(ListContactsRequest(
            name = null, // Set to Some("Alice") to filter by name
            offset = 0u,
            limit = 10u
        ))
        for (contact in contacts) {
            // Log.v("Breez", "Contact: id=${contact.id}, name=${contact.name}, identifier=${contact.paymentIdentifier}")
        }
        // ANCHOR_END: list-contacts
    }
}

File: docs/breez-sdk/snippets/react-native/contacts.ts
import { type BreezSdk } from '@breeztech/breez-sdk-spark-react-native'

const exampleAddContact = async (sdk: BreezSdk) => {
  // ANCHOR: add-contact
  const contact = await sdk.addContact({
    name: 'Alice',
    paymentIdentifier: 'alice@example.com'
  })
  console.log(`Contact added: ${JSON.stringify(contact)}`)
  // ANCHOR_END: add-contact
}

const exampleUpdateContact = async (sdk: BreezSdk) => {
  // ANCHOR: update-contact
  const contactId = 'contact-id'
  const contact = await sdk.updateContact({
    id: contactId,
    name: 'Alice Smith',
    paymentIdentifier: 'alice.smith@example.com'
  })
  console.log(`Contact updated: ${JSON.stringify(contact)}`)
  // ANCHOR_END: update-contact
}

const exampleDeleteContact = async (sdk: BreezSdk) => {
  // ANCHOR: delete-contact
  const contactId = 'contact-id'
  await sdk.deleteContact(contactId)
  console.log('Contact deleted')
  // ANCHOR_END: delete-contact
}

const exampleListContacts = async (sdk: BreezSdk) => {
  // ANCHOR: list-contacts
  // List contacts with pagination (e.g., 10 contacts starting from offset 0)
  // Optionally filter by exact name match
  const contacts = await sdk.listContacts({
    name: undefined, // Set to "Alice" to filter by name
    offset: 0,
    limit: 10
  })
  for (const contact of contacts) {
    console.log(`Contact: id=${contact.id}, name=${contact.name}, identifier=${contact.paymentIdentifier}`)
  }
  // ANCHOR_END: list-contacts
}
